pkg/provider/azure_loadbalancer.go

```diff
@@ -1763,6 +1766,21 @@ func (az *Cloud) reconcileLoadBalancer(clusterName string, service *v1.Service,
 	return lb, nil
 }
 
+// addOrUpdateLBInList adds or updates the given lb in the list
+func addOrUpdateLBInList(lbs *[]network.LoadBalancer, targetLB *network.LoadBalancer) {
+	var found bool
+	for i, lb := range *lbs {
+		if strings.EqualFold(pointer.StringDeref(lb.Name, ""), pointer.StringDeref(targetLB.Name, "")) {
+			(*lbs)[i] = *targetLB
+			found = true
+			break
```

Comment:
here we can return and in line 1780 we can append item to array. Do you think It will be easier if we put all of the results into a map?

Reply:
Do you mean to remove `found` and use return to replace `break`? And what map do you mean?

Reply:
yes.

Reply:
I'm wondering if map[string]*network.Loadbalaner meet your needs instead of *[]network.LoadBalancer

Reply:
prefer just return instead of break

Reply:
done